### PR TITLE
Fixes lp#1742955: Add trace to pprof.

### PR DESCRIPTION
**Pull request by anastasiamac**
## Description of change

As per the linked bu, this may help diagnose some production conditions...

## Bug reference

https://bugs.launchpad.net/juju/+bug/1742955

### CODE REVIEW

**Comment by axinojolais** [comment]
LGTM, although I haven't tested the code itself. Thanks !

File: worker/introspection/pprof/pprof.go [hunk 1]
```diff
@@ -163,6 +166,44 @@ func Symbol(w http.ResponseWriter, r *http.Request) {
 	w.Write(buf.Bytes())
 }
 
+func durationExceedsWriteTimeout(r *http.Request, seconds float64) bool {
+	srv, ok := r.Context().Value(http.ServerContextKey).(*http.Server)
+	return ok && srv.WriteTimeout != 0 && seconds >= srv.WriteTimeout.Seconds()
```
Comment (wallyworld) [inline worker/introspection/pprof/pprof.go]: technically, exceeds means > rather than >=

Comment (anastasiamac) [reply worker/introspection/pprof/pprof.go]: Sure :D but as a direct copy from go 1.9, I did not want to introduce too much difference.

**Comment by anastasiamac** [comment]
$$merge$$

**Pull request merged**